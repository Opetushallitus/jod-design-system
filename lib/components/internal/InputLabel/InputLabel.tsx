import { tidyClasses as tc } from '../../../utils';

interface InputLabelProps {
  /** The id of the form element that this label is associated with */
  htmlFor: string;
  /** The text to display inside the label */
  labelText?: string;
  /** Hide label */
  hideLabel?: boolean;
}

export const InputLabel = ({ htmlFor, labelText, hideLabel = false }: InputLabelProps) => {
  return (
    <label
      htmlFor={htmlFor}
      className={tc([
        hideLabel ? 'ds:hidden' : 'ds:inline-block',
        'ds:mb-4 ds:align-top ds:text-form-label ds:font-arial ds:text-primary-gray',
      ])}
    >
      {labelText}
    </label>
  );
};
