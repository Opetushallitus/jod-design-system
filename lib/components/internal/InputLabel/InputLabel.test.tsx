import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { InputLabel } from './InputLabel';

describe('InputLabel', () => {
  const labelText = 'Test Label';
  const htmlFor = 'test-input';

  it('renders label correctly', () => {
    const { getByText, container } = render(<InputLabel htmlFor={htmlFor} labelText={labelText} />);
    expect(getByText('Test Label')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('hides label when hideLabel is true', () => {
    const { getByText, container } = render(<InputLabel htmlFor={htmlFor} labelText={labelText} hideLabel={true} />);
    const labelElement = getByText('Test Label');
    expect(labelElement).toBeInTheDocument();
    expect(labelElement).toHaveClass('ds:hidden');
    expect(container).toMatchSnapshot();
  });
  it('applies the provided htmlFor to the label element', () => {
    const { getByText, container } = render(<InputLabel htmlFor={htmlFor} labelText={labelText} />);
    expect(getByText('Test Label')).toHaveAttribute('for', htmlFor);
    expect(container).toMatchSnapshot();
  });
  it('renders without labelText', () => {
    const { container } = render(<InputLabel htmlFor={htmlFor} />);
    expect(container).toMatchSnapshot();
  });
});
