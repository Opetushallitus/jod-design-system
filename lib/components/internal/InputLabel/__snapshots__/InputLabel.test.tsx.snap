// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputLabel > applies the provided htmlFor to the label element 1`] = `
<div>
  <label
    class="ds:inline-block ds:mb-4 ds:align-top ds:text-form-label ds:font-arial ds:text-primary-gray"
    for="test-input"
  >
    Test Label
  </label>
</div>
`;

exports[`InputLabel > hides label when hideLabel is true 1`] = `
<div>
  <label
    class="ds:hidden ds:mb-4 ds:align-top ds:text-form-label ds:font-arial ds:text-primary-gray"
    for="test-input"
  >
    Test Label
  </label>
</div>
`;

exports[`InputLabel > renders label correctly 1`] = `
<div>
  <label
    class="ds:inline-block ds:mb-4 ds:align-top ds:text-form-label ds:font-arial ds:text-primary-gray"
    for="test-input"
  >
    Test Label
  </label>
</div>
`;

exports[`InputLabel > renders without labelText 1`] = `
<div>
  <label
    class="ds:inline-block ds:mb-4 ds:align-top ds:text-form-label ds:font-arial ds:text-primary-gray"
    for="test-input"
  />
</div>
`;
